src/libs/elektra/keyset.c

```diff
@@ -852,6 +905,7 @@ ssize_t ksAppendKey (KeySet * ks, Key * toAppend)
 			ks->array[insertpos] = toAppend;
 			ksSetCursor (ks, insertpos);
 		}
+		elektraOpmphmInvalidate (ks);
```

Comment:
Also add it to ksAppend in the case we optimize it?

Reply:
I wrote a test case for this scenario, that test case will remind the `ksAppend (....)` optimizer that there is something missing.

Reply:
Another reason is that not all ksAppends lead to an invalidation and there is no way to find out in the actual implementation of ksAppend.

Reply:
Thank you for writing this testcase, this is the best solution!